hasktorch/src/Torch/Tensor.hs

```diff
@@ -202,6 +205,82 @@ instance TensorLike [[[Double]]] where
   asTensor v = mkTensor @Float (map realToFrac (concat (concat v))) [length v, length (head v), length (head (head v))] float_opts
   asValue t = split2d (shape t) $ map realToFrac $ mkValue @Float t
 
+--------------------------------------------------------------------------------
+-- Tuple -> ForeignPtr (ATen's Tuple)
+--------------------------------------------------------------------------------
+
+instance Castable (Tensor,Tensor) (ForeignPtr (ATen.Tensor,ATen.Tensor)) where
+  cast _ _ = undefined
+  uncast t f = do
+    t0 <- get0 t
+    t1 <- get1 t
+    f (Unsafe t0, Unsafe t1)
+
+instance Castable (Tensor,Tensor,Tensor) (ForeignPtr (ATen.Tensor,ATen.Tensor,ATen.Tensor)) where
```

Comment:
Can't we generate instances like `(Castable a a', Castable b b', Castable c c') => Castable (a, b, c) (a', b', c')` as part of the ATen codegen?

Reply:
Yes, we can. I think so.
In addition to this, we can generate simple funtions like gels.

Reply:
For now, I am thinking how to implement `(Castable a a', Castable b b', Castable c c') => Castable (a, b, c) (a', b', c')`.

Reply:
The implementation is done.